test/src/ProducerTest.cc

```diff
@@ -0,0 +1,58 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one or more
+* contributor license agreements.  See the NOTICE file distributed with
+* this work for additional information regarding copyright ownership.
+* The ASF licenses this file to You under the Apache License, Version 2.0
+* (the "License"); you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+#ifndef WIN32
+#include <unistd.h>
+#endif
+#include <stdio.h>
+
+#include "CProducer.h"
+#include "CCommon.h"
+#include "CMessage.h"
+#include "CSendResult.h"
+#include "gtest/gtest.h"
+
+TEST(Producer, producer) {
+    printf("Producer Initializing.....\n");
+
+    CProducer* producer = CreateProducer("Group_producer");
+    SetProducerNameServerAddress(producer,"172.17.0.2:9876");
```

Comment:
Don't use a real address `"172.17.0.2:9876"` in a test.
Just use `localhost` instead of it.